channeldb/kvdb/interface.go

```diff
@@ -0,0 +1,90 @@
+package kvdb
+
+import (
+	"github.com/btcsuite/btcwallet/walletdb"
+	_ "github.com/btcsuite/btcwallet/walletdb/bdb" // Import to register backend.
+)
+
+// Update opens a database read/write transaction and executes the function f
+// with the transaction passed as a parameter. After f exits, if f did not
+// error, the transaction is committed. Otherwise, if f did error, the
+// transaction is rolled back. If the rollback fails, the original error
+// returned by f is still returned. If the commit fails, the commit error is
+// returned.
+var Update = walletdb.Update
+
+// View opens a database read transaction and executes the function f with the
+// transaction passed as a parameter. After f exits, the transaction is rolled
+// back. If f errors, its error is returned, not a rollback error (if any
+// occur).
+var View = walletdb.View
```

Comment:
I'm unsure how this could be cleanly implemented in `etcd` the same way as we use `View` all around the code. There's no concept of `View` in `etcd` as there's no such thing as read transaction. What we have instead is a transaction where given conditions we can execute range queries (gets) and then retrieve the result.